worlds/yoshisisland/Rules.py

```diff
@@ -0,0 +1,597 @@
+from .level_logic import YoshiLogic
+from worlds.generic.Rules import set_rule
+
+
+
+logic: YoshiLogic
+def set_easy_rules(world, multiworld, player: int, boss_order, luigi_pieces):
+    logic = YoshiLogic(world)
+    set_rule(multiworld.get_location("Make Eggs, Throw Eggs: Red Coins", player), lambda state: state.has_all({'Dashed Stairs', 'Beanstalk'}, player))
+    set_rule(multiworld.get_location("Make Eggs, Throw Eggs: Flowers", player), lambda state: state.has_all({'Dashed Stairs', 'Beanstalk'}, player))
+    set_rule(multiworld.get_location("Make Eggs, Throw Eggs: Stars", player), lambda state: state.has_all({'Tulip', 'Beanstalk', 'Dashed Stairs'}, player))
+    set_rule(multiworld.get_location("Make Eggs, Throw Eggs: Level Clear", player), lambda state: state.has_all({'Beanstalk'}, player))
+
+    set_rule(multiworld.get_location("Watch Out Below!: Red Coins", player), lambda state: state.has_all({'Large Spring Ball', 'Helicopter Morph'}, player))
+    set_rule(multiworld.get_location("Watch Out Below!: Flowers", player), lambda state: state.has_all({'Large Spring Ball', 'Helicopter Morph'}, player))
+    set_rule(multiworld.get_location("Watch Out Below!: Stars", player), lambda state: state.has_all({'Large Spring Ball'}, player) and logic.has_midring(state))
+    set_rule(multiworld.get_location("Watch Out Below!: Level Clear", player), lambda state: state.has_all({'Large Spring Ball'}, player))
+
+    set_rule(multiworld.get_location("The Cave Of Chomp Rock: Red Coins", player), lambda state: state.has_all({'Chomp Rock'}, player))
+    set_rule(multiworld.get_location("The Cave Of Chomp Rock: Flowers", player), lambda state: state.has_all({'Chomp Rock'}, player))
+
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Red Coins", player), lambda state: state.has_all({'Spring Ball'}, player))
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Flowers", player), lambda state: state.has_all({'Spring Ball', 'Key'}, player) and (state.has('Egg Capacity Upgrade', player, 3) or logic.combat_item(state)))
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Stars", player), lambda state: state.has_all({'Spring Ball'}, player) and (logic.has_midring(state) or state.has('Key', player)))
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Level Clear", player), lambda state: logic._14CanFightBoss(state))
+    set_rule(multiworld.get_location("Burt The Bashful Defeated", player), lambda state: state.has_all({'Egg Plant'}, player))

```

Comment:
"Burt The Bashful Defeated" is not a location in this world. The intended one was probably the "Burt the Bashful's Boss Room" location, But also "Burt the Bashful Defeated" is a better name, so would recommend changing the location name instead on line 39 of Locations.py

Reply:
Oh, this wasn't supposed to be present in rules